### PR TITLE
Minor fixes and version 0.2.0

### LINKED ISSUE #317: Any reason to use devtools::install_github over remotes::install_github
The devtools folks created the [remotes](https://github.com/r-lib/remotes) package as a lighter-weight successor for simply doing installs like `install_github()`. Is there any reason for continued use of `devtools` in the `cmdstanr` guides? (ditto for the rest of the Stan-iverse?)

### LINKED ISSUE #338: Add a check that the user has permission in the install and temporary folder
Basically what the title says. We can add this to the toolchain check.

Reported by user franzsf on Discourse.

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

- fixes #338 
- fixes #317
- removes some code that was used for simpler handling of turning threading/mpi/opencl on off pre-2.24. I think we can safely remove that.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
Rok Češnovar, Uni. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: R/install.R [hunk 1]
```diff
@@ -427,11 +427,16 @@ build_status_ok <- function(process_log, quiet = FALSE) {
 }
 
 install_mingw32_make <- function(quiet = FALSE) {
+  rtools_usr_bin <- file.path(Sys.getenv("RTOOLS40_HOME"), "usr", "bin")
+  if (!checkmate::test_directory(rtools_usr_bin, access = "w")) {
+    warning("No write permissions in the RTools folder. This might prevent installing mingw32-make."
+            " Consider changing permissions or reinstalling RTools in a different folder.", call. = FALSE)
```
Comment (rok-cesnovar) [inline R/install.R:433]: This is only a warning because things might work regardless.

File: R/model.R [hunk 2]
```diff
@@ -468,16 +470,13 @@ compile_method <- function(quiet = TRUE,
   stanc_built_options = c()
   for (i in seq_len(length(stanc_options))) {
     option_name <- names(stanc_options)[i]
-    if (isTRUE(as.logical(stanc_options[[i]]))) {
+    if (isTRUE(as.logical(stanc_options[[i]])) || is.null(option_name)) {
```
Comment (rok-cesnovar) [inline R/model.R:473]: This is so that we also allow stanc_options = list("allow-undefined")

File: R/utils.R [hunk 3]
```diff
@@ -208,64 +208,20 @@ cpp_options_to_compile_flags <- function(cpp_options) {
   paste0(cpp_built_options, collapse = " ")
 }
 
-prepare_precompiled <- function(cpp_options = list(), quiet = FALSE) {
```
Comment (rok-cesnovar) [inline R/utils.R:211]: This was only being used for 2.23 and earlier and it's mostly a power user thing and not essential to work with cmdstan so I think we can remove this. For users that want seamless switching of threads/mpi/opencl on and off we should recommend switching to 2.25.

**Comment by jgabry** [comment]
I haven't done the 0.2.0 release yet, so we might as well merge this first. I'll take a look now.

**Comment by rok-cesnovar** [comment]
There is nothing major here, but we might as well get this in.

File: R/utils.R [hunk 4]
```diff
@@ -208,64 +208,20 @@ cpp_options_to_compile_flags <- function(cpp_options) {
...
-                 file.path("stan", "src", "stan", "model", "model_header.hpp.gch")),
-        wd = cmdstan_path(),
-        echo_cmd = !quiet,
-        echo = !quiet,
-        spinner = quiet,
-        stderr_line_callback = function(x,p) { if (!quiet) message(x) },
-        error_on_status = TRUE
-      )
-      file.copy(model_header_path_gch, model_header_path_w_flags)
+check_stanc_options <- function(stanc_options) {
```
Comment (jgabry) [inline R/utils.R:211]: Is the `i` (and the `i <- i + 1` below) used in this function necessary? Can't we just replace all `i` with the `s` that's in the for loop? Or am I missing something (totally possible)?

File: tests/testthat/test-model-compile.R [hunk 5]
```diff
@@ -295,4 +295,51 @@ test_that("pedantic check works", {
...
+})
 
+test_that("compiling works with only names in list", {
+  skip_on_cran()
+  stan_file <- testing_stan_file("bernoulli")
+  mod <- cmdstan_model(stan_file, stanc_options = list("allow-undefined", "warn-pedantic"), force_recompile = TRUE)
+  expect_type(
+    mod,
+    "dada"
+  )
```
Comment (jgabry) [inline tests/testthat/test-model-compile.R:344]: What's "dada"?

**Comment by jgabry** [comment]
Just put two tiny questions in review comments but otherwise this looks good!

File: tests/testthat/test-model-compile.R (hunk 5 above)
Comment (rok-cesnovar) [reply tests/testthat/test-model-compile.R:344]: Oh lol, dummy text didnt fix :)

Comment (rok-cesnovar) [reply tests/testthat/test-model-compile.R:344]: its yesyes in slovenian :D

Comment (jgabry) [reply tests/testthat/test-model-compile.R:344]: Haha, nice!

File: R/utils.R (hunk 4 above)
Comment (rok-cesnovar) [reply R/utils.R:211]: Hm, I need the `i` for the names. Not sure how else I can get the name from the list value? I am still an R noob.

File: tests/testthat/test-model-compile.R [hunk 6]
```diff
@@ -295,4 +295,51 @@ test_that("pedantic check works", {
...
+    mod$compile(stanc_options = hyphens3),
+    "No leading hyphens allowed in stanc options (--o). Use options without leading hyphens, like for example `stanc_options = list('allow-undefined')`",
+    fixed = TRUE
+  )
+})
 
+test_that("compiling works with only names in list", {
+  skip_on_cran()
+  stan_file <- testing_stan_file("bernoulli")
+  mod <- cmdstan_model(stan_file, stanc_options = list("allow-undefined", "warn-pedantic"), force_recompile = TRUE)
```
Comment (jgabry) [inline tests/testthat/test-model-compile.R:340]: I'm getting an error when I run this line

File: R/utils.R (hunk 4 above)
Comment (jgabry) [reply R/utils.R:211]: I might be misunderstanding but aren't `i` and `s` always the same value in this for loop? If so then all `i` could be replaced with `s`. But maybe I'm being an idiot ;)

Comment (rok-cesnovar) [reply R/utils.R:211]: Basically for 
```
list(
"test",
"test1" = TRUE
)
```
s would be
test and TRUE but I need the name in the second case.

Comment (jgabry) [reply R/utils.R:211]: Oh I'm an idiot. `s` isn't an integer at all it's the stanc option!

Comment (jgabry) [reply R/utils.R:211]: So just ignore this, what you have is fine!

File: tests/testthat/test-model-compile.R (hunk 6 above)
Comment (jgabry) [reply tests/testthat/test-model-compile.R:340]: Error message is 

```
bin/stanc: unknown option '--=allow-undefined'.
```

So it looks like it's putting the `=` in between the `--` and the option name

File: R/utils.R (hunk 4 above)
Comment (jgabry) [reply R/utils.R:211]: Yeah I was reading it too quickly and assuming s was an index but it's the option itself.

File: tests/testthat/test-model-compile.R (hunk 6 above)
Comment (rok-cesnovar) [reply tests/testthat/test-model-compile.R:340]: Should be fixed now.

Comment (rok-cesnovar) [reply tests/testthat/test-model-compile.R:340]: Thanks!

Comment (jgabry) [reply tests/testthat/test-model-compile.R:340]: Yeah working now, thanks!

**Comment by jgabry** [comment]
Since this branch is just minor fixes I'm just going to use this same branch for the 0.2.0 stuff (updating DESCRIPTION, regenerating online doc, etc.). That way when we merge it I can just tag 0.2.0 and we'll be all set.

**Comment by rok-cesnovar** [comment]
Sure. Oh, while I have you here. Can you post the exact commands you run to regenerate the website? Just curious, I always fail with that :)

**Comment by jgabry** [comment]
If I'm regenerating the entire site (I'll do that this time) then this: 

```r
withr::with_envvar(c("NOT_CRAN" = "true"), pkgdown::build_site(run_dont_run = TRUE))
```

If I just need to regenerate a single or a few doc pages then this: 

```r
# for example just to update the doc for check_syntax
withr::with_envvar(c("NOT_CRAN" = "true"), {
  pkgdown::build_reference(topics = c("model-method-check_syntax"), run_dont_run = TRUE)
})
```

Without the `NOT_CRAN="true"` stuff the code chunks won't be executed in the vignettes. That's because I added stuff to the vignettes to make sure they're not evaluated on CRAN (since CRAN won't have cmdstan). But I guess I could have waited to add that until we were actually going to submit to CRAN and then the withr stuff wouldn't be necessary right now.

**Comment by rok-cesnovar** [comment]
Thanks!

**Pull request merged**